src/target/riscv/riscv-013.c

```diff
@@ -3109,47 +3102,69 @@ static int read_sbcs_nonbusy(struct target *target, uint32_t *sbcs)
 }
 
 /* TODO: return mem_access_result_t */
-static int modify_privilege(struct target *target, uint64_t *mstatus, uint64_t *mstatus_old)
+static int modify_privilege(struct target *target, riscv_reg_t *mstatus, riscv_reg_t *mstatus_old,
```

Comment:
```suggestion
static int modify_privilege_for_virt2phys_mode(struct target *target, riscv_reg_t *mstatus, riscv_reg_t *mstatus_old,
```

Up to your consideration, I recommend renaming this function to reflect what it does more. `modify_privilege()` feels too generic to me. Or maybe you have a better idea for a name.

Reply:
Addressed